Tella/Scenes/Dropbox/ViewModel/DropboxOutboxViewModel.swift

```diff
@@ -0,0 +1,217 @@
+//
+//  DropboxOutboxViewModel.swift
+//  Tella
+//
+//  Created by gus valbuena on 9/19/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class DropboxOutboxViewModel: OutboxMainViewModel<DropboxServer> {
+    let dropboxRepository: DropboxRepositoryProtocol
+    private var cancellables = Set<AnyCancellable>()
+    
+    override var shouldShowCancelUploadConfirmation: Bool {
+        return true
+    }
+    
+    init(reportsViewModel: ReportsMainViewModel,
+                  reportId: Int?,
+                  repository: DropboxRepositoryProtocol) {
+        self.dropboxRepository = repository
+        super.init(reportsViewModel: reportsViewModel, reportId: reportId)
+        
+        if reportViewModel.status == .submissionScheduled {
+            self.submitReport()
+        } else {
+            self.pauseSubmission()
+        }
+    }
+    
+    override func initVaultFile(reportId: Int?) {
+        if let reportId, let report = self.mainAppModel.tellaData?.getDropboxReport(id: reportId) {
+            
+            let vaultFileResult  = mainAppModel.vaultFilesManager?.getVaultFiles(ids: report.reportFiles?.compactMap{$0.fileId} ?? [])
+            
+            var files: [ReportVaultFile] = []
+            
+            report.reportFiles?.forEach({ reportFile in
+                guard let vaultFile = vaultFileResult?.first(where: {reportFile.fileId == $0.id}),
+                      let dropboxReportFile = reportFile as? DropboxReportFile else { return}
+                let reportVaultFile = ReportVaultFile(reportFile: dropboxReportFile, vaultFile: vaultFile)
+                files.append(reportVaultFile)
+            })
+            
+            self.reportViewModel = ReportViewModel(report: report, files: files)
+            
+        }
+    }
+    
+    
+    override func submitReport() {
+        if isSubmissionInProgress { return }
+
+        self.updateReportStatus(reportStatus: .submissionInProgress)
+        cancellables.removeAll()
+
+        if reportViewModel.folderId != nil {
```

Comment:
I think it's better to move the upload handling to the `DropboxRepository` and handling the result in the view model. 
There is an example  in `NextcloudOutboxViewModel` : `performSubmission()` function